src/dashboard/src/components/archival_storage/views.py

```diff
@@ -121,101 +235,94 @@ def search(request):
+            augmented_results = search_augment_file_results(es_client, results)
+        else:
+            augmented_results = search_augment_aip_results(results, uuid_file_counts)
+
+        return helpers.json_response(
+            {
+                "iTotalRecords": hit_count,
+                "iTotalDisplayRecords": hit_count,
+                "sEcho": int(
+                    request.GET.get("sEcho", 0)
+                ),  # It was recommended we convert sEcho to int to prevent XSS
+                "aaData": augmented_results,
+            }
+        )
 
     except ElasticsearchException:
-        logger.exception("Error accessing index.")
-        return HttpResponse("Error accessing index.")
-
-    if not file_mode:
-        aic_creation_form = forms.CreateAICForm(initial={"results": uuids})
-    else:  # if file_mode
-        aic_creation_form = None
-
-    page_data = helpers.pager(results, items_per_page, current_page_number)
-
-    return render(
-        request,
-        "archival_storage/search.html",
-        {
-            "file_mode": file_mode,
-            "show_aics": show_aics,
-            "checked_if_in_file_mode": checked_if_in_file_mode,
-            "aic_creation_form": aic_creation_form,
-            "results": page_data.object_list,
-            "search_params": search_params,
-            "page": page_data,
-        },
-    )
+        err_desc = "Error accessing AIPs index"
+        logger.exception(err_desc)
+        return HttpResponse(err_desc)
 
 
-def _get_es_field(record, field, default=None):
-    if field not in record or not record[field]:
-        return default
-    return record[field]
+def search_augment_aip_results(raw_results, counts):
+    """Augment AIP results and check and update ES status if AIP is pending deletion.
 
+    We perform check_and_update_aip_pending_deletion routine here to avoid
+    needing to iterate through all of the results a second time.
 
-def search_augment_aip_results(raw_results, counts):
+    :param raw_results: Raw results returned from ES
+    :param counts: Count of file UUIDs associated with AIP
+
+    :return: Augmented and formatted results
+    """
     modified_results = []
 
     for item in raw_results["hits"]["hits"]:
         fields = item["_source"]
         new_item = {
-            "name": fields["name"],
-            "uuid": fields["uuid"],
-            "count": counts[fields["uuid"]],
-            "created": fields["created"],
-            "isPartOf": _get_es_field(fields, "isPartOf"),
-            "AICID": _get_es_field(fields, "AICID"),
-            "countAIPsinAIC": _get_es_field(fields, "countAIPsinAIC", "(unknown)"),
-            "type": "AIC" if "AIC#" in _get_es_field(fields, "AICID", "") else "AIP",
-            "status": AIP_STATUS_DESCRIPTIONS[
-                _get_es_field(fields, "status", "UPLOADED")
-            ],
-            "encrypted": _get_es_field(fields, "encrypted", False),
+            "name": fields.get("name"),
```

Comment:
This function is taking values from the ES index (which does handle null values okay - though if we have a missing name, I think that would be an indication of a larger issue!) and formatting them before returning them to the frontend. The frontend does expect a string, so that would make a better default!